### ISSUE TITLE
misleading: 400 kHz is Fast speed, not  Low Speed !

### ISSUE THREAD

**Issue by shiftleftplusone** [SMFSW/WireWrapper#1]
hello,
why do you call 400 kHz Low Speed?
I2C_init(I2C_LOW); // init with low speed (400KHz)

the common definitions for i2c bus speed are the following:
Standard Mode (Sm) 	100 kbit/s 	bidirektional
Fast Mode (Fm) 	400 kbit/s 	bidirektional
Fast Mode Plus (Fm+) 	1,0 Mbit/s 	bidirektional
High Speed Mode (Hs-mode) 	3,4 Mbit/s 	bidirektional
Ultra Fast-mode (UFm) 	5,0 Mbit/s 	unidirektional

so low speed should be used just for clock speed slower  than standard,  
400 kHz should be called I2C_FAST
and
100 kHz should be called I2C_STD

OTOH.
if there once sth like 9.6 kHz would be provided or needed, then you might call that  I2C_LOW instead

**Comment by SMFSW** [comment]
Hello,

alright, I'll change enum names for the different speeds.

I'll post an update later in the day.

Thanks for your feedback!

**Comment by shiftleftplusone** [comment]
now I see, thank you very much for considering the changes!
kind regards!

**Comment by SMFSW** [comment]
You're welcomed, I wasn't happy with the names anyways!

Well, i'll take one more day before committing the changes as I didn't had time yet to test the changes, and I will try to give more explanations in readmes for both WireWrapper and cI2C libs.

By the way, new names would be:
- I2C_STD: standard speed 100KHz
- I2C_FM: fast mode 400KHz
- I2C_FMP: fast mode plus 1MHz
- & added I2C_HS: High speed 3.4MHz

I dropped Ultra Fast Mode as it's unidirectional, and also 9.6KHz as it doesn't seems to be that relevant.

I made an enum of speeds so that it can be used instead of giving an integer as parameter for bus Init function, but in particular cases, an integer can be passed as argument.
The only limitation yet is that value given is clamped between STD (400K) and HS (3.4M).
If you think it would be better not to clamp the value, let me know.

Thanks for your interest, it's really appreciated!
Best regards!
